src/components/Inputs/Select.tsx

```diff
@@ -0,0 +1,65 @@
+import { forwardRef, useState, useCallback, type DetailedHTMLProps, type SelectHTMLAttributes } from "react";
+import { twJoin } from "tailwind-merge";
+import { RiArrowDownSLine } from "react-icons/ri";
+import { Portal } from "../Portal";
+import "./Select.css";
+
+export interface SelectOption {
+  value: string;
+  label: string;
+}
+
+export interface SelectProps
+  extends Omit<DetailedHTMLProps<SelectHTMLAttributes<HTMLDivElement>, HTMLDivElement>, "value" | "onChange"> {
+  options: SelectOption[];
+  value?: string;
+  onChange?: (value: string) => void;
+  className?: string;
+  placeholder?: string;
+}
+
+export const Select = forwardRef<HTMLDivElement, SelectProps>(
+  ({ className, options, value, onChange, placeholder = "Select option", ...props }, ref) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [selectedOption, setSelectedOption] = useState<SelectOption | undefined>(
```

Comment:
This component doesn't support uncontrolled state. Use `useControlledState` hook instead